### PR TITLE
Enemy AI w/ patrolling and attack

### PR DESCRIPTION
**Pull request by redknight990**
- Added enemy prefab & script
- Can now add enemies to the scene w/ customizable params such as speed, vision angle, patrol route
- Enemies will attack player if in vision
- Player can lose enemies by breaking line-of-sight

### CODE REVIEW

File: Hush/Assets/Plugins/Line Of Sight/Scripts/LOSObjectHider.cs
Comment (arejayelle) [inline Hush/Assets/Plugins/Line Of Sight/Scripts/LOSObjectHider.cs:61]: Could we just get the renderers in `Start()`, store it, and then just check there instead of re-getting them every update? It doesn't seems like the renderer ever changes during runtime

**Review by arejayelle — COMMENTED** [review]
Looks great! Just a small comment for @po-trottier for performance stuff. 

I'll approve after Dany's code is merged tho, since your branch is based on hers

File: Hush/Assets/Plugins/Line Of Sight/Scripts/LOSObjectHider.cs
Comment (po-trottier) [reply Hush/Assets/Plugins/Line Of Sight/Scripts/LOSObjectHider.cs:61]: Yeah! Didn't wanna modify the Plugin file too much but that's worth a change! @redknight990 do you think you could do it?

Comment (arejayelle) [reply Hush/Assets/Plugins/Line Of Sight/Scripts/LOSObjectHider.cs:61]: Ah didn't realize it was part of a plugin. If it's not our code, I don't care.

**Review by arejayelle — APPROVED** [review]
Awesome! Took me a sec to figure out how to set a patrol route. Very sexy. 10/10 🐸

**Pull request merged**
